src/app/zeko/da-layer/da_utils/da_utils.ml

```diff
@@ -0,0 +1,58 @@
+(** 
+  These functions call external C functions that call rust functions defined in rust library
+  These are long running functions with IO operations, so they are run in separate threads,
+  C code calls `caml_release_runtime_system` so that other threads can run in parallel.    
+*)
```

Comment:
We probably don't want to be using threads here in the future FWIW. You'd have a FD (e.g. pipe) you listen on.

Reply:
How is it better? You create the thread anyways, and this follows the standard way of doing the FFI.

Reply:
WDYM you create the thread anyway? There are various ways of doing this, but one simple way is creating a (pipe) FD, passing it to Rust-side with FFI, and Rust-side giving a new (pipe) FD back. You write to FD your JSON, close it. You `select`/`epoll`/etc. on FD you got back. `async_unix` supports this IIRC.

Reply:
Some thread needs to write the JSON to the FD.

Reply:
the one that runs the rust code

Reply:
Rust OS thread != OCaml green thread